app/graphql/resolvers/create_article.rb

```diff
@@ -15,17 +19,26 @@ class Resolvers::CreateArticle < Resolvers::MutationFunction
   # args - are the arguments passed
   # _ctx - is the GraphQL context (which would be discussed later)
   def call(_obj, args, ctx)
-    validate_admin(ctx)
+    if !admin_is_valid(ctx)
+      return GraphQL::ExecutionError.new("Invalid user token. Please log in.")
+    end
     @article = Article.new(
```

Comment:
This whole thing needs to be in a transaction. That way if stuff fails, it won't leave the db in a malformed state.

Reply:
So you want me to surround everything after the first `if` statement in a `ActiveRecord::Base.transaction do ... end` ?

Reply:
yep pretty much